### PR TITLE
Bugfix/zcs 5109

### PR DESCRIPTION
**Pull request by rupalid**
Added code to read the attributes from LDAP.
Replaced existing code to use LdapConfiguration class

### CODE REVIEW

File: src/java/com/zimbra/oauth/utilities/LdapConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+     * @param appName
+     * @throws ServiceException
+     */
+    protected LdapConfiguration(String appName) {
+        super(appName);    
+    }
+    
+    
+    /**
+     *
```
Comment (jdavid-synacor) [inline src/java/com/zimbra/oauth/utilities/LdapConfiguration.java]: Remove empty doc block.

File: src/java/com/zimbra/oauth/utilities/LdapConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+     */
+    @Override
+    public String getString(String key) {
+        
+        String appName = getAppName(key);
+        return getConfig(key, appName);
+    }
+
+    /**
+     * @param oauthClientId
```
Comment (jdavid-synacor) [inline src/java/com/zimbra/oauth/utilities/LdapConfiguration.java]: A brief description of, method, params & return.

File: src/java/com/zimbra/oauth/utilities/OAuth2Constants.java [hunk 3]
```diff
@@ -67,5 +67,16 @@
...
     public static final String LC_OAUTH_SCOPE_TEMPLATE = "zm_oauth_%s_scope";
     public static final String LC_OAUTH_IMPORT_CLASS_TEMPLATE = "zm_oauth_%s_import_class";
+    
+    
+    public static final String OAUTH_CLIENT_ID = "client_id";
+    public static final String OAUTH_CLIENT_SECRET = "client_secret";
+    public static final String OAUTH_CLIENT_REDIRECT_URI = "client_redirect_uri";
+    public static final String OAUTH_SCOPE = "scope";
+
+    public static final String APPNAME_YAHOO = "yahoo";
```
Comment (desouzas) [inline src/java/com/zimbra/oauth/utilities/OAuth2Constants.java]: These constants exist at `YahooConstants.CLIENT_NAME`, `GoogleConstants.CLIENT_NAME`, `FacebookConstants.CLIENT_NAME`, and `OutlookConstants.CLIENT_NAME`.

File: src/java/com/zimbra/oauth/utilities/LdapConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+    }
+    
+    
+    /**
+     * 
+     */
+    @Override
+    public String getString(String key) {
+        
+        String appName = getAppName(key);
```
Comment (desouzas) [inline src/java/com/zimbra/oauth/utilities/LdapConfiguration.java]: The property `clientId` on the parent class, `Configuration`, already knows the app name. You don't need to determine it here. Just make `clientId` `protected` then you can just use it directly, or call the public method `this.getClientId`, in the `getConfig` method.

File: src/java/com/zimbra/oauth/utilities/LdapConfiguration.java [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+    }
+
+
+    /**
+     * Return the name of the client app, based on the key.
+     * @param key the client related key(zm_oauth_yahoo_client_id,zm_oauth_google_client_secret,
+     * zm_oauth_outlook_client_redirect_uri
+     * @return the client app name
+     */
+    private String getAppName(String key) {
```
Comment (desouzas) [inline src/java/com/zimbra/oauth/utilities/LdapConfiguration.java]: The property `clientId` from the parent class already knows which social service this instance of the (ldap)configuration class is for, so you don't need this method. Just make clientId `protected` and use it directly, or call the public method `this.getClientId`.

File: src/java/com/zimbra/oauth/utilities/LdapConfiguration.java [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+
+import com.zimbra.common.service.ServiceException;
+import com.zimbra.common.util.ZimbraLog;
+import com.zimbra.cs.account.Provisioning;
+
+/**
+ * The Configuration class for this project for loading attributes from LDAP
+ * @author zimbra
+ *
+ */
```
Comment (desouzas) [inline src/java/com/zimbra/oauth/utilities/LdapConfiguration.java]: For consistency:

```
/**
 * The LdapConfiguration class for this project.
 *
 * @author Zimbra API Team
 * @package com.zimbra.oauth.utilities
 * @copyright Copyright © 2018
 */
```

File: src/java/com/zimbra/oauth/utilities/LdapConfiguration.java (hunk 1 above)
Comment (rupalid) [reply src/java/com/zimbra/oauth/utilities/LdapConfiguration.java]: Done

File: src/java/com/zimbra/oauth/utilities/LdapConfiguration.java (hunk 2 above)
Comment (rupalid) [reply src/java/com/zimbra/oauth/utilities/LdapConfiguration.java]: Done

**Comment by rupalid** [comment]
Done

File: src/java/com/zimbra/oauth/utilities/LdapConfiguration.java (hunk 6 above)
Comment (rupalid) [reply src/java/com/zimbra/oauth/utilities/LdapConfiguration.java]: Done

File: src/java/com/zimbra/oauth/utilities/LdapConfiguration.java (hunk 5 above)
Comment (rupalid) [reply src/java/com/zimbra/oauth/utilities/LdapConfiguration.java]: Fixed

File: src/java/com/zimbra/oauth/utilities/LdapConfiguration.java (hunk 4 above)
Comment (rupalid) [reply src/java/com/zimbra/oauth/utilities/LdapConfiguration.java]: Fixed

**Pull request merged**